src/components/implementation/cbufp/default/cbufp.c

```diff
@@ -381,12 +378,14 @@ cbufp_create(spdid_t spdid, int size, long cbid)
 	 * Update the meta with the correct addresses and flags!
 	 */
 	memset(meta, 0, sizeof(struct cbuf_meta));
-	meta->nfo.c.flags |= CBUFM_TOUCHED | 
-		             CBUFM_OWNER  | CBUFM_WRITABLE;
-	meta->nfo.c.ptr    = cbi->owner.addr >> PAGE_ORDER;
-	meta->sz           = cbi->size >> PAGE_ORDER;
-	if (meta->nfo.c.refcnt == CBUFP_REFCNT_MAX) assert(0);
-	meta->nfo.c.refcnt++;
+	CBUF_SET_OWNER(meta);
+	CBUF_SET_WRITABLE(meta);
+	CBUF_SET_TOUCHED(meta);
```

Comment:
This is not how flags work.  You're relying on the compiler to do a lot of optimizations to not make this horribly slow.  It is reasonable that it will in many cases, but I'm not sure why we care to make it work hard in this case.

More importantly, everyone knows how to use bit fields for flags, and how to modify them.  So why break that convention and intuition there?  Better might be:

`CBUF_FLAGS_ADD(CBUFM_TOUCHED | CBUFM_OWNER  | CBUFM_WRITABLE);`

and, if you wish, a `CBUF_FLAGS_REM(...)` corresponding call.

The `CBUFM_PTR_SET(...)` is fine, but you might as well add the same for size and cbid.

Please remember the naming conventions.  You always try for `noun_verb`, not `verb_noun`.  So when you have `SET_PTR`, you should really have `PTR_SET` (same for REFCNT, etc....).